lib/build_manager.py

```diff
@@ -33,7 +33,8 @@ def __init__(self, packages_names, distro):
     def __call__(self):
```

Comment:
Yes, I'm assuming there will be only one shared repository for each package for all builds. This avoids having to clone potentially big repositories multiple times. The actual build can be executed in parallel, since after the archive is ready the repository is not used anymore. I'm adding this improvement in a separate commit.
I'm improving the commit message.